rcl/test/rcl/test_graph.cpp

```diff
@@ -1223,75 +1223,175 @@ TEST_F(CLASSNAME(TestGraphFixture, RMW_IMPLEMENTATION), test_graph_query_functio
-  std::promise<bool> topic_changes_promise;
-  std::thread topic_thread(
-    [this, &topic_changes_promise]() {
-      // sleep
-      std::this_thread::sleep_for(std::chrono::milliseconds(200));
-      // create the publisher
-      rcl_publisher_t pub = rcl_get_zero_initialized_publisher();
-      rcl_publisher_options_t pub_ops = rcl_publisher_get_default_options();
-      rcl_ret_t ret = rcl_publisher_init(
-        &pub, this->node_ptr, ROSIDL_GET_MSG_TYPE_SUPPORT(test_msgs, msg, BasicTypes),
-        "/chatter_test_graph_guard_condition_topics", &pub_ops);
-      EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
-      // sleep
-      std::this_thread::sleep_for(std::chrono::milliseconds(200));
-      // create the subscription
-      rcl_subscription_t sub = rcl_get_zero_initialized_subscription();
-      rcl_subscription_options_t sub_ops = rcl_subscription_get_default_options();
-      ret = rcl_subscription_init(
-        &sub, this->node_ptr, ROSIDL_GET_MSG_TYPE_SUPPORT(test_msgs, msg, BasicTypes),
-        "/chatter_test_graph_guard_condition_topics", &sub_ops);
-      EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
-      // sleep
-      std::this_thread::sleep_for(std::chrono::milliseconds(200));
-      // destroy the subscription
-      ret = rcl_subscription_fini(&sub, this->node_ptr);
-      EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
-      // sleep
-      std::this_thread::sleep_for(std::chrono::milliseconds(200));
-      // destroy the publication
-      ret = rcl_publisher_fini(&pub, this->node_ptr);
-      EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
-      // notify that the thread is done
-      topic_changes_promise.set_value(true);
-    });
-  // Wait for the graph state to change, expecting it to do so at least 4 times,
-  // once for each change in the topics thread.
+  std::chrono::nanoseconds time_to_sleep = std::chrono::milliseconds(400);
+
+  rcl_wait_set_t wait_set = rcl_get_zero_initialized_wait_set();
+  ret = rcl_wait_set_init(
+    &wait_set, 0, 1, 0, 0, 0, 0, context_ptr, rcl_get_default_allocator());
+  ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+  {
+    EXPECT_EQ(RCL_RET_OK, rcl_wait_set_fini(&wait_set)) << rcl_get_error_string().str;
+  });
+
   const rcl_guard_condition_t * graph_guard_condition =
-    rcl_node_get_graph_guard_condition(this->node_ptr);
-  ASSERT_NE(nullptr, graph_guard_condition) << rcl_get_error_string().str;
-  std::shared_future<bool> future = topic_changes_promise.get_future();
-  size_t graph_changes_count = 0;
-  // while the topic thread is not done, wait and count the graph changes
-  while (future.wait_for(std::chrono::seconds(0)) != std::future_status::ready) {
-    ret = rcl_wait_set_clear(this->wait_set_ptr);
+    rcl_node_get_graph_guard_condition(node_ptr);
+
+  // Wait for no graph change condition
+  int idx = 0;
+  while (idx < 100) {
```

Comment:
where is `++idx`?

maybe a `for` loop is more clear

Reply:
Sorry. It's a mistake. will fix it.